rfcs/0107-version-normalization.md

```diff
@@ -0,0 +1,182 @@
+- Release denotes any distributed snapshot, as defined by its team.
+- Branch denotes a logical sequence of snapshots, as identified by the program's
+  team;
+  - Usually these branches are denoted by names such like `stable`, `master`,
+    `unstable`, `trunk`, `experimental`, `staging` etc.;
+
+## Design
+
+- For a labeled snapshot:
+  - `version` should be constituted of the version of the snapshot, as defined
+    by the upstream project, without any alphabetical characters (e.g. "v",
+    "rel") prepending it.
+    - Alphabetical characters after the first should be maintained, except
+      optionally those clearly used as separators, in which case they are
+      replaced by dots (emulating a typical semver).
+
+- For an unlabeled snapshot:
+  - `version` should be constituted of a concatenation of the elements below in
+    this order:
+      - the version of the latest labeled snapshot, as defined above;
+        - If the project never released a labeled snapshot, `0_0` should be used
+          as default.
+     - the string `+unstable=YYYY-MM-DD`, where `YYYY-MM-DD` denotes the date
+       the mentioned unlabeled snapshot was distributed.
+
+- Some atypical programs have special considerations:
+  - Linux kernel modules:
+    - Besides the rules established above for typical snapshots (whether labeled
+      or unlabeled), `version` shoud have appended `+linux=XX.XX.XX`, where
+      `XX.XX.XX` is the corresponding Linux kernel version used to build the
+      aforementioned module.
+
+# Examples and Interactions
+[examples-and-interactions]: #examples-and-interactions
+
+Some useful examples:
+
+- Bochs is a typical Sourceforge-hosted project; its labeled snapshots can be
+  fetched from tarballs obtained via URLs like
+  <https://sourceforge.net/projects/bochs/files/bochs/2.6.11/>; for this
+  example, we have `pname = "bochs"; version = "2.6.11";`.
+
+- MPV is a typical Github-hosted program; its labeled snapshots can be fetched
+  from tarballs obtained via URLs like
+  <https://github.com/mpv-player/mpv/releases/tag/v0.33.1>; for this example, we
+  get rid of the `"v"` prepended to the version tag: `pname = "mpv"; version =
+  "0.33.1";`.
+
+- SDL2 is hosted on Github; its latest labeled version can be downloaded from
+  <https://github.com/libsdl-org/SDL/releases/tag/release-2.0.14>; therefore we
+  have `pname = "SDL2"; version = "2.0.14";`.
+
+  _However_, this labeled version was released December 21, 2020, while the
+  latest change was done in May 28, 2021. Therefore, for this particular
+  unlabeled releases of SDL2, we have `pname = "SDL2"; version =
+  "2.0.14+unstable=2021-05-28";`.
+
+- Cardboard is a typical Gitlab-hosted program; it has no labeled release yet,
+  therefore we use `0.0.0` as default stable version; the latest commit was made
+  on May 10, 2021; therefore, we have `pname = "cardboard"; version =
+  "0_0+unstable=2021-05-21";`.
+
```

Comment:
Maybe add a mention of PostgreSQL or Squid or some other package with multiple major-versioned brances?

Reply:
Good. It can be useful to explain this particular case before composing it into text.

Reply:
I have used Python as example